sshd-core/src/main/java/org/apache/sshd/client/channel/ChannelSession.java

```diff
@@ -217,4 +223,42 @@ protected int securedRead(
             }
         }
     }
+
+    /**
+     * @param  key   The (never {@code null}) key (Note: may be empty...)
+     * @param  value The value to set - if {@code null} then the pre-existing value for the key (if any) is
+     *               <U>removed</U>.
+     * @return       The replaced/removed previous value - {@code null} if no previous value set for the key.
+     */
+    public Object setEnv(String key, Object value) {
+        ValidateUtils.checkNotNull(key, "No key provided");
+        if (value == null) {
+            return env.remove(key);
+        } else {
+            return env.put(key, value);
+        }
+    }
+
+    protected void sendEnvVariables(Session session) throws IOException {
+        if (MapEntryUtils.size(env) > 0) {
+            if (log.isDebugEnabled()) {
+                log.debug("Sending env variables ({}) Send SSH_MSG_CHANNEL_REQUEST env: {}", this, env);
+            }
+
+            // Cannot use forEach because of the IOException being thrown by writePacket
+            for (Map.Entry<String, ?> entry : env.entrySet()) {
+                String key = entry.getKey();
+                Object value = entry.getValue();
+                String str = Objects.toString(value);
+                Buffer buffer = session.createBuffer(
+                        SshConstants.SSH_MSG_CHANNEL_REQUEST, key.length() + GenericUtils.length(str) + Integer.SIZE);
+                buffer.putInt(getRecipient());
+                buffer.putString("env");
+                buffer.putBoolean(false); // want-reply
```

Comment:
The standard does not say that *wantReply* must be *false* - while I do agree that it is a sensible default have you given thought how yo handle *true* ? At the very least this choice should be documented and perhaps open a follow-up issue to add support for *wantReply=true* in some future release